docs/agents.md

```diff
@@ -1,128 +1,233 @@
-# Agents
+# Agent
+
+The `BaseAgent` class is the foundation of the Bee Framework, providing the core interface and functionality that all agent implementations must follow. It orchestrates the interaction between LLMs, tools, memory, and development utilities to create intelligent, automated workflows.
+
+## Overview
+
+`BaseAgent` acts as an abstract base class that defines the standard interface and basic functionality for all agents in the framework. It manages the lifecycle of agent operations, coordinates between different components, and provides a consistent interface for agent implementations.
+
+## Architecture
+
+```mermaid
+classDiagram
+    class BaseAgent {
+        +LLM llm
+        +Memory memory
+        +Tool[] tools
+        +DevTools devTools
+        +run(prompt: string, options?: ExecutionOptions)
```

Comment:
This is an invalid signature. The first parameter is `input`, which is generic—it is up to the person who implements it to define it.

Reply:
I think that I mixing some diagrams. Let me update it with the current implemented version and open a discussion about evolutions in the architecture.